.github/workflows/ci-go.yml

```
@@ -0,0 +1,46 @@
+# Continuous integration handling for Go
+name: ci-go
+
+on:
+  pull_request:
+    paths:
+      - .github/workflows/test.yml
+      - .go-version
+      - go.mod
+      - '**.go'
+
+permissions:
+  contents: read
+
+jobs:
+  golangci-lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - id: go-version
+        # Reference: https://github.com/actions/setup-go/issues/23
+        run: echo "::set-output name=version::$(cat ./.go-version)"
+      - uses: actions/setup-go@v2
+        with:
+          go-version: ${{ steps.go-version.outputs.version }}
+      - run: go mod download
+      - uses: golangci/golangci-lint-action@v2
+        with:
+          skip-go-installation: true
+  test:
```

GitHub Actions will attempt to make a CI check name that combines the job name and any matrix strategy combinations, but you could also specify your own if you wish to add some extra readability to CI checks (totally optional):

```suggestion
  test:
    name: "Test (Go v${{ matrix.go }}"
```

---

I love this and add it. 👍